plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java

```diff
@@ -196,17 +198,40 @@ private String requestCompletion(
 
             try {
                 List<?> choices;
+                int responseSize = GPT_MODEL_MAX_RESPONSE_TOKENS;
                 for (int i = 0; ; i++) {
                     try {
                         choices = getCompletionChoices(service, completionRequest);
                         break;
                     } catch (Exception e) {
-                        if (e instanceof HttpException && ((HttpException) e).code() == 429) {
-                            RuntimeUtils.pause(1000);
+                        if ((e instanceof HttpException && ((HttpException) e).code() == 429)
+                            || (e instanceof OpenAiHttpException && e.getMessage().contains("This model's maximum"))) {
+                            if (e instanceof HttpException) {
+                                RuntimeUtils.pause(1000);
+                            } else {
+                                // Extracts resulted prompt size from the error message and resizes max response to
+                                // value lower that (maxTokens - prompt size)
+                                Matcher matcher = Pattern.compile("([0-9]+)").matcher(e.getMessage());
```

Comment:
Use proper regexp which matches error message. Current one just searches for any numbers.
If error message doesn't match expected one then do not try to recalculate tokens.